org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -3365,6 +3354,31 @@ private void findCompletionsForArgumentPosition(ObjectVector methodsFound, int c
 		}
 	}
 
+	private void pushExpectedTypesForArgumentPosition(ObjectVector methodsToSearchOn, int completedArgumentLength,
+			Scope scope) {
+		if (methodsToSearchOn.size == 0) {
+			return;
+		}
+		for (int i = 0; i < methodsToSearchOn.size; i++) {
+			MethodBinding method = (MethodBinding) ((Object[]) methodsToSearchOn.elementAt(i))[0];
+			boolean onOrAfterLastParam = method.parameters.length <= completedArgumentLength;
+			if (onOrAfterLastParam && !method.isVarargs()) {
+				continue;
+			}
+
+			int index = onOrAfterLastParam
+					? method.parameters.length - 1
+					: completedArgumentLength;
+
+			TypeBinding paramType = method.parameters[index];
+			addExpectedType(paramType, scope);
+			if (method.isVarargs() && paramType.isArrayType()
+					&& index == (method.parameters.length - 1)) {
```

Comment:
`onOrAfterLastParam` can be used here as well?

Reply:
I think so even thought the meaning of this check is bit different. But looking at how the index is calculated using the same make sense.

Reply:
Well a test proves we cannot, the reason is when we have not filled any arguments, `onOrAfterLastParam` is false, but still we want `index == (method.parameters.length - 1)` true if the method only has a vararg.